app/uk/gov/hmrc/customs/emailfrontend/connectors/Save4LaterConnector.scala

```diff
@@ -26,89 +26,103 @@ import uk.gov.hmrc.customs.emailfrontend.connectors.http.responses.{BadRequest,
-        None
-    }
+    val getUrl = s"${appConfig.save4LaterUrl}/$id/$key"
+    http.get(url"$getUrl")
+      .execute[ReferrerName]
+      .map { response =>
+        auditCallResponse[ReferrerName](getUrl, response)
+        Some(response)
+      }.recover {
+        case e => logger.error(s"Unable to get Referrer :${e.getMessage}")
+          None
+      }
   }
 
   def getJourneyType(id: String, key: String)
                     (implicit hc: HeaderCarrier, reads: Reads[JourneyType]): Future[Option[JourneyType]] = {
 
-    val url = s"${appConfig.save4LaterUrl}/$id/$key"
-    http.GET[JourneyType](url).map { response =>
-      auditCallResponse[JourneyType](url, response)
-      Some(response)
-    }.recover {
-      case e => logger.error(s"Unable to get journey type :${e.getMessage}")
-        None
-    }
+    val getUrl = s"${appConfig.save4LaterUrl}/$id/$key"
+    http.get(url"$getUrl")
+      .execute[JourneyType]
+      .map { response =>
+        auditCallResponse[JourneyType](getUrl, response)
+        Some(response)
+      }.recover {
+        case e => logger.error(s"Unable to get journey type :${e.getMessage}")
+          None
+      }
   }
 
   def put[T](id: String, key: String, payload: JsValue)
             (implicit hc: HeaderCarrier): Future[Either[HttpErrorResponse, Unit]] = {
 
-    val url = s"${appConfig.save4LaterUrl}/$id/$key"
-    logger.info(s"PUT: $url")
-    auditCallRequest(url, payload)
-
-    http.PUT[JsValue, HttpResponse](url, payload).map { response =>
-      auditCallResponse(url, response.status)
-      response.status match {
-        case NO_CONTENT | CREATED | OK => Right(())
-        case _ => Left(BadRequest)
+    val putUrl = s"${appConfig.save4LaterUrl}/$id/$key"
+    logger.info(s"PUT: $putUrl")
+    auditCallRequest(putUrl, payload)
+
+    http.put(url"$putUrl")
+      .withBody[JsValue](payload)
+      .execute[HttpResponse]
+      .map { response =>
+        auditCallResponse(putUrl, response.status)
+        response.status match {
+          case NO_CONTENT | CREATED | OK => Right(())
+          case _ => Left(BadRequest)
+        }
+      }.recover {
+        case e => logger.error(s"Request failed for call to $putUrl, exception: ${e.getMessage}")
+          Left(UnhandledException)
       }
-    }.recover {
-      case e => logger.error(s"Request failed for call to $url, exception: ${e.getMessage}")
-        Left(UnhandledException)
-    }
   }
 
   def delete[T](id: String)(implicit hc: HeaderCarrier): Future[Either[HttpErrorResponse, Unit]] = {
 
-    val url = s"${appConfig.save4LaterUrl}/$id"
-    logger.info(s"DELETE: $url")
-    auditCallRequest(url, JsNull)
-
-    http.DELETE[HttpResponse](url).map { response =>
-      auditCallResponse(url, response.status)
-      response.status match {
-        case NO_CONTENT => Right(())
-        case _ => Left(BadRequest)
+    val deleteUrl = s"${appConfig.save4LaterUrl}/$id"
+    logger.info(s"DELETE: $deleteUrl")
```

Comment:
Better to give a line break before this line to separate logger from the url and to make it more readable